bellatrix.plugins.visual-regression-tracker/src/main/java/solutions/bellatrix/plugins/vrt/VisualRegressionService.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2024 Automate The Planet Ltd.
+ * Author: Miriam Kyoseva
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package solutions.bellatrix.plugins.vrt;
+
+import io.visual_regression_tracker.sdk_java.TestRunOptions;
+import io.visual_regression_tracker.sdk_java.TestRunResult;
+import io.visual_regression_tracker.sdk_java.VisualRegressionTracker;
+import io.visual_regression_tracker.sdk_java.VisualRegressionTrackerConfig;
+import plugins.screenshots.ScreenshotPlugin;
+import solutions.bellatrix.core.configuration.ConfigurationService;
+import solutions.bellatrix.core.utilities.SingletonFactory;
+
+import java.io.IOException;
+import java.util.Objects;
+
+public class VisualRegressionService {
+    private static float defaultDiffTolerance() {
+        return ConfigurationService.get(VisualRegressionSettings.class).getDefaultDiffTolerance();
+    }
+
+    private static final ThreadLocal<VisualRegressionTracker> VISUAL_REGRESSION_TRACKER_THREAD_LOCAL;
+    private static final ThreadLocal<VisualRegression> VISUAL_REGRESSION_ATTRIBUTE;
+
+    static {
+        try {
+            VISUAL_REGRESSION_TRACKER_THREAD_LOCAL = new ThreadLocal<>();
```

Comment:
`ThreadLocal.withInitial(() -> null);` This is an alternative approach to achieving the same result. The key distinction is that it employs a static factory method, which is the standard convention for object creation in the Core Java API.